lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -612,66 +602,44 @@ def get_mount_facts(self):
                     mount_info['options'] += ",bind"
 
             results[mount] = {'info': mount_info, 'timelimit': time.time() + maxtime}
-            if pool is None:
-                old_handler = signal.signal(signal.SIGALRM, _timeout_handler)
-                signal.alarm(maxtime)
-                try:
-                    size, uuid = self.get_mount_info(mount, device, uuids)
-                except TimeoutError as e:
-                    results[mount]['info']['note'] = 'Could not get extra information due to timeout'
-                    self.module.log(f"Timeout while gathering mount {mount} data: {e}")
-                    self.module.warn(f"Timeout exceeded when getting mount info for {mount}")
-                finally:
-                    signal.alarm(0)
-                    signal.signal(signal.SIGALRM, old_handler)
-
-                if size:
-                    results[mount]['info'].update(size)
-                results[mount]['info']['uuid'] = uuid or 'N/A'
-            else:
-                # use multiproc pool, handle results below
-                results[mount]['extra'] = pool.apply_async(self.get_mount_info, (mount, device, uuids))
+            results[mount]['extra'] = pool.apply_async(self.get_mount_info, (mount, device, uuids))
 
-        if pool is None:
-            # serial processing, just assing results
-            mounts.append(results[mount]['info'])
-        else:
             pool.close()  # done with spawing new workers, start gc
```

Comment:
The pool needs to be closed outside the for loop:

```suggestion
        pool.close()  # done with spawing new workers, start gc
```